leapp/reporting/__init__.py

```diff
@@ -0,0 +1,191 @@
+from leapp.models import Model, fields
+from leapp.topics import ReportTopic
+from leapp.libraries.stdlib.api import produce
+import json
+
+
+class ReportVersions(Model):
+    topic = ReportTopic
+
+    framework = fields.String()
+    actors = fields.String()
+
+
+class Renderers(Model):
+    topic = ReportTopic
+
+    html = fields.String()
+    plaintext = fields.String()
+
+
+class Report(Model):
+
+    __non_inheritable__ = True
+
+    topic = ReportTopic
+
+    """
+    Version of framework and actors as :class:`ReportVersions`
+    """
+    versions = fields.Model(ReportVersions)
+
+    """
+    Severity of the report entry
+    """
+    severity = fields.StringEnum(choices=['low', 'medium', 'high'])
+
+    """
+    Title of the report entry
+    """
+    title = fields.String()
+
+    """
+    Detail of the report entry as JSON data
+    """
+    detail = fields.String()
+
+    """
+    Named variables, represented as JSON data, that can be used during rendering
+
+    Example:
+        named = {'summary_with_exclamations': '!!!!! { summary } !!!!'}
+    """
+    named = fields.Nullable(fields.String())
```

Comment:
I thought 'detail' was supposed to be just that? I'm a bit confused about this field.

Reply:
I just realized that we're not processing the field correctly. Nevertheless, the main difference should be:
1) `detail` is not jinja2 processed
2) `named` is jinja2 processed
I.e. in `detail` I don't need to worry about escaping possible embedded jinja2 strings, it remains static.
I'm fine with removing `named` and reintroduce it when it's needed as this is not really crucial for the enablement.